spec/models/user_spec.rb

```diff
@@ -13,6 +13,31 @@
     it { is_expected.to validate_presence_of(:provider) }
 
     it { is_expected.to validate_presence_of(:email) }
+    it { is_expected.to validate_presence_of(:password_confirmation) }
     it { is_expected.to validate_uniqueness_of(:email).scoped_to(:provider).case_insensitive }
+
+    context 'password confirmation' do
+      let(:valid_user) do
+        {
+          name: Faker::Name.name,
+          email: Faker::Internet.email,
+          provider: 'greenlight',
+          password: 'Password123+',
+          password_confirmation: 'Password123+'
+        }
+      end
+
+      it 'validate the record for confirmed password' do
+        user = described_class.new valid_user
+        expect(user).to be_valid
+      end
+
+      it 'invalidate the record for mismatched password confirmation' do
+        user = described_class.new valid_user
+        user.password_confirmation = "#{user.password}extra"
+        expect(user).to be_invalid
+        expect(user.errors.first.details).to eq({ error: :confirmation, attribute: 'Password' })
```

Comment:
Use the factory here instead. 

```
user = build(:user, password_confirmation: "....")
expect(user).to be_invalid
expect(user.errors.first.details).to eq({ error: :confirmation, attribute: 'Password' })
```

Reply:
Didn't know about the `build` method.
Thank you, I'll change it.

Reply:
Done,